File: TieBomber.java
import javax.swing.*;
import java.awt.*;
import java.util.Random;

/** Tie bomber class for jWars project */
public class TieBomber extends Enemy {
	private ImageIcon imgIcon;

	/** standard constructor takes the highest possible y position on the screen
	 *  @param lowestYpos lowest y position */
	public TieBomber(int lowestYpos) {
		super(lowestYpos);
		this.setPointValue(10);
		imgIcon = new ImageIcon("tiebomber.gif");
		setEntityWidth(imgIcon.getIconWidth());
		setEntityHeight(imgIcon.getIconHeight());
	}

	/** Paint the tie bomber on the screen
	 *  @param g graphics component to paint on */
	public void paintComponent(Graphics g) {
		if(getXpos() == -1 && getYpos() == -1) {
			Random generator = new Random(System.nanoTime());
			setXpos(generator.nextInt(600) + 1);
			generator = new Random(System.nanoTime());
			setYpos(generator.nextInt(299) + 1);	
		}

		g.drawImage(imgIcon.getImage(), getXpos(), getYpos(), null, null);
	}
}

File: src/TieBomber.scala
import javax.swing._
import java.awt._
import scala.util.Random

/** Tie bomber class for jWars project */
class TieBomber(lyp: Int) extends Enemy(lyp) {

	pointValue = 10

	private val imgIcon = new ImageIcon("tiebomber.gif")
	entityWidth = imgIcon.getIconWidth()
	entityHeight = imgIcon.getIconHeight()

	/** Paint the tie bomber on the screen
	 *  @param g graphics component to paint on */
	override def paintComponent(g: Graphics) = {
		if(xPos == -1 && yPos == -1) {
			xPos = Random.nextInt(600) + 1
			yPos = Random.nextInt(299) + 1
		}

		g.drawImage(imgIcon.getImage(), xPos, yPos, null, null)
		updateLocations()
	}
}
